backend/api/v1/auth_service.go

```diff
@@ -189,26 +189,12 @@ func (s *AuthService) CreateUser(ctx context.Context, request *v1pb.CreateUserRe
 		Type:         principalType,
 		PasswordHash: string(passwordHash),
 	}
-	if request.User.UserRole != v1pb.UserRole_USER_ROLE_UNSPECIFIED {
-		rolePtr := ctx.Value(common.RoleContextKey)
-		// Allow workspace owner to create user with role.
-		if rolePtr != nil && rolePtr.(api.Role) == api.WorkspaceAdmin {
-			userRole := convertUserRole(request.User.UserRole)
-			if userRole == api.UnknownRole {
-				return nil, status.Errorf(codes.InvalidArgument, "invalid user role %s", request.User.UserRole)
-			}
-			userMessage.Role = userRole
-			userMessage.Roles = append(userMessage.Roles, userRole)
-			for _, r := range request.User.GetRoles() {
-				role, err := common.GetRoleID(r)
-				if err != nil {
-					return nil, status.Errorf(codes.InvalidArgument, "invalid role %s", r)
-				}
-				userMessage.Roles = append(userMessage.Roles, api.Role(role))
-			}
-		} else {
-			return nil, status.Errorf(codes.PermissionDenied, "only workspace owner can create user with role")
+	for _, role := range request.User.Roles {
+		roleID, err := common.GetRoleID(role)
+		if err != nil {
+			return nil, status.Errorf(codes.InvalidArgument, err.Error())
 		}
+		userMessage.Roles = append(userMessage.Roles, api.Role(roleID))
```

Comment:
Need to check that only the workspaceAdmin can create users with roles other than workspaceMember.

Reply:
Use `common.UserContextKey` to get the creator user and inspect `user.Roles`.

Reply:
Done